src/engine/EventManager.cxx

```diff
@@ -399,29 +399,21 @@ void EventManager::checkEvents(SDL_Event &event, Engine &engine)
             m_nodesToPlace = m_nodesToHighlight;
           }
 
-          // we need to check if placement is allowed and set a bool to color ALL the highlighted tiles and not just those who can't be placed
+          // Finally highlight all the tiles we've found
+          // Set highlighted tiles that can be placed and can't be placed different color
+          m_placementAllowed = false;
           for (const auto &highlitNode : m_nodesToHighlight)
           {
             if (!engine.map->isPlacementOnNodeAllowed(highlitNode, tileToPlace) || demolishMode)
             {
-              // already occupied tile, mark red
-              m_placementAllowed = false;
-              break;
-            }
-
-            // mark gray.
-            m_placementAllowed = true;
-          }
-          // finally highlight all the tiles we've found
-          for (const auto &highlitNode : m_nodesToHighlight)
-          {
-            if (m_placementAllowed)
-            {
-              engine.map->highlightNode(highlitNode, SpriteHighlightColor::GRAY);
+              // mark red
```

Comment:
did you check these changes with buildings bigger than 1 tile ? 
we did twice loop here because we want all the buildings tiles to be marked as red when placement isn't allowed, I think as I see from the new implementation that the buildings bigger that 1x1 won't be all of them highlighted as red but only the not allowed area. @SimplyLiz  what do you think ?

Reply:
![grafik](https://user-images.githubusercontent.com/7029836/167816992-142507a0-16c6-4e39-9a98-b6532a6dd5fc.png)

This would allow placement of buildings on water, you are right @Mograbi

Reply:
Note to myself. We need unittests for this